### DISCUSSION TITLE
Calling a function after updating the state

### DISCUSSION THREAD

**Discussion by tomtobac** [pmndrs/zustand#2440] [Q&A] [answered]
Hey there 👋 

I am sorry if this has been answered or explained. I couldn't find any reference, so I'm doing something wrong. 

How can I use a callback after updating the store? The store itself says that it is updated, but the function looks like it's bonded with the previous value

Repo example: https://codesandbox.io/p/sandbox/zustand-problem-xhu5sj

```
new value is "1"
calling api with the name value ""
new value is "12"
calling api with the name value "1"
```

**Comment by dai-shi** [comment, marked as answer]
It depends on your requirement, but for example:
```js
    console.log(`calling api with the name value "${useStore.getState().name}"`);
```

But, I would prefer this:
```ts
  const makeApiCall = (name: string) => {
    console.log(`calling api with the name value "${name}"`);
  };
```

**Reply by tomtobac** [reply]
oh, thanks @dai-shi ! 🙇 

I was looking into `useStore.getState()` but I couldn't because my store is typed manually since I am using `persist` with `skipHydration: true`.

Do you know how to type this kind of `useStore`? 

```ts
export type StoreState = ChatStore & AnotherStore;

const _useStore = create<StoreState>()(
  persist(
    (...a) => ({
      ...createChatSlice(...a),
      ...createAnotherSlice(...a),
    }),
    {
      name: "store",
      version: 1,
      skipHydration: true,
    }
  )
);

type UseStore = <T>(selector: (state: StoreState) => T) => T;

export const useStore: UseStore = (selector) => {
  useEffect(() => {
    _useStore.persist.rehydrate();
  }, []);

  return _useStore(selector);
};

```

**Reply by dai-shi** [reply]
You don't need to attach to the hook.
```js
const { getState } = _useStore;
```
would be simple.
